File: Project/java/src/test/java/com/basanta/todo/TaskStatusTest.java
package com.basanta.todo;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class TaskStatusTest {

    @Test
    void allStatusesIncludesPendingAndCompleted() {
        assertTrue(TaskStatus.ALL_STATUSES.contains(TaskStatus.PENDING));
        assertTrue(TaskStatus.ALL_STATUSES.contains(TaskStatus.COMPLETED));
        assertEquals(3, TaskStatus.ALL_STATUSES.size());
    }

    @Test
    void isValidStatusAcceptsValid() {
        assertTrue(TaskStatus.isValidStatus("PENDING"));
        assertTrue(TaskStatus.isValidStatus("IN_PROGRESS"));
    }

    @Test
    void isValidStatusRejectsInvalid() {
        assertFalse(TaskStatus.isValidStatus("DONE"));
        assertFalse(TaskStatus.isValidStatus(null));
    }
}

File: Project/javaScript/tests/task-status.test.js
/**
 * Tests for TaskStatus module.
 */
import { describe, it } from 'node:test';
import assert from 'node:assert';
import { ALL_STATUSES, isValidStatus, PENDING, COMPLETED } from '../src/TaskStatus.js';

describe('TaskStatus', () => {
  it('ALL_STATUSES includes PENDING and COMPLETED', () => {
    assert.ok(ALL_STATUSES.includes(PENDING));
    assert.ok(ALL_STATUSES.includes(COMPLETED));
    assert.strictEqual(ALL_STATUSES.length, 3);
  });

  it('isValidStatus accepts valid statuses', () => {
    assert.strictEqual(isValidStatus('PENDING'), true);
    assert.strictEqual(isValidStatus('IN_PROGRESS'), true);
    assert.strictEqual(isValidStatus('COMPLETED'), true);
  });

  it('isValidStatus rejects invalid values', () => {
    assert.strictEqual(isValidStatus(''), false);
    assert.strictEqual(isValidStatus('DONE'), false);
    assert.strictEqual(isValidStatus(null), false);
  });
});
